packages/block-library/src/embed/embed-link-settings.native.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	LinkSettingsNavigation,
+	FooterMessageLink,
+} from '@wordpress/components';
+import { isURL } from '@wordpress/url';
+import { useDispatch } from '@wordpress/data';
+import { store as noticesStore } from '@wordpress/notices';
+import { useCallback, useRef, useState } from '@wordpress/element';
+
+const EmbedLinkSettings = ( {
```

Comment:
This component is a copy of the `EmbedBottomSheet` component but with the following tweaks:

### URL is now kept in two places
The URL is now kept as an instance variable via the `useRef` and in the component state. The reason for this is that when `onDismiss` is called, I realized that the previous `url` state value ([reference](https://github.com/WordPress/gutenberg/pull/36099/files?authenticity_token=Ah3%2FJt4s4h5hECy91JtulYIvCsOA1prp%2BgPsIMxjD23WBg3FzYJhCbA8f0qyAKOE%2BlHAyPqav9wNMjPuwFlWIQ%3D%3D&file-filters%5B%5D=.js&file-filters%5B%5D=.md#diff-7d155d7b9eb26681fd43d964649784c4799d33b187271b3288e604f9c9728b8fL15)) isn't updated even after the `setURL` call ([reference](https://github.com/WordPress/gutenberg/pull/36099/files?authenticity_token=Ah3%2FJt4s4h5hECy91JtulYIvCsOA1prp%2BgPsIMxjD23WBg3FzYJhCbA8f0qyAKOE%2BlHAyPqav9wNMjPuwFlWIQ%3D%3D&file-filters%5B%5D=.js&file-filters%5B%5D=.md#diff-7d155d7b9eb26681fd43d964649784c4799d33b187271b3288e604f9c9728b8fL60)). Not sure why is this happening but I decided to keep the internal value of the URL as an instance value, which is used in the different handlers, and have a URL in the component state only for passing it to the `LinkSettingsNavigation` component.

### Autofocus prop
The link field shouldn't be autofocused when editing an URL via the block settings, so I added a prop to control it.

### onDismiss handler simplified
I noticed that the code could be simplified as there was an `else` case that was redundant ([reference](https://github.com/WordPress/gutenberg/pull/36099/files?authenticity_token=Ah3%2FJt4s4h5hECy91JtulYIvCsOA1prp%2BgPsIMxjD23WBg3FzYJhCbA8f0qyAKOE%2BlHAyPqav9wNMjPuwFlWIQ%3D%3D&file-filters%5B%5D=.js&file-filters%5B%5D=.md#diff-7d155d7b9eb26681fd43d964649784c4799d33b187271b3288e604f9c9728b8fL53-L56)).